mantidimaging/gui/windows/main/view.py

```diff
@@ -559,31 +580,36 @@ def closeEvent(self, event):
             # Ignore the close event, keeping window open
             event.ignore()
 
-    def cleanup(self):
+    def cleanup(self) -> None:
```

Comment:
@samtygier-stfc The shared_array attribute to None directly is not allowed because the shared_array property is read-only. I suggest a clear_shared_array method in the ImageStack class that clears the shared memory.

Reply:
May need to use an ignore here:
`stack.shared_array = None # type: ignore # Only happens when cleaning up`